projects/com.oracle.truffle.llvm.asm.amd64/src/com/oracle/truffle/llvm/asm/amd64/AsmFactory.java

```diff
@@ -0,0 +1,199 @@
+                break;
+            case "orl":
+                this.statements.add(LLVMWriteI32NodeGen.create(LLVMAMD64OrlNodeGen.create(leftNode, rightNode), rightSlot));
+                break;
+            case "xorl":
+                this.statements.add(LLVMWriteI32NodeGen.create(LLVMAMD64XorlNodeGen.create(leftNode, rightNode), rightSlot));
+                break;
+            case "sall":
+                this.statements.add(LLVMWriteI32NodeGen.create(LLVMAMD64SallNodeGen.create(leftNode, rightNode), rightSlot));
+                break;
+            case "sarl":
+                this.statements.add(LLVMWriteI32NodeGen.create(LLVMAMD64SarlNodeGen.create(leftNode, rightNode), rightSlot));
+                break;
+            case "shll":
+                this.statements.add(LLVMWriteI32NodeGen.create(LLVMAMD64ShllNodeGen.create(leftNode, rightNode), rightSlot));
+                break;
+            case "shrl":
+                this.statements.add(LLVMWriteI32NodeGen.create(LLVMAMD64ShrlNodeGen.create(leftNode, rightNode), rightSlot));
+                break;
+            case "movl":
+                this.statements.add(LLVMWriteI32NodeGen.create(LLVMAMD64MovlNodeGen.create(leftNode, rightNode), rightSlot));
+                break;
+            default:
+                this.statements.add(new LLVMI32UnsupportedInlineAssemblerNode());
+        }
+    }
+
+    public void createUnaryOperation(String operation, String node) {
+        FrameSlot slot = frameDescriptor.findFrameSlot(node);
+        LLVMI32Node childNode = LLVMI32ReadNodeGen.create(slot);
+        this.result = LLVMI32ReadNodeGen.create(slot);
+        switch (operation) {
+            case "notl":
+                this.statements.add(LLVMWriteI32NodeGen.create(LLVMAMD64NotlNodeGen.create(childNode), slot));
+                break;
+            case "decl":
+                this.statements.add(LLVMWriteI32NodeGen.create(LLVMAMD64DeclNodeGen.create(childNode), slot));
+                break;
+            case "incl":
+                this.statements.add(LLVMWriteI32NodeGen.create(LLVMAMD64InclNodeGen.create(childNode), slot));
+                break;
+            default:
+                this.statements.add(new LLVMI32UnsupportedInlineAssemblerNode());
+                break;
+        }
+    }
+
+    protected void addFrameSlot(String reg, LLVMBaseType type) {
+        if (!registers.contains(reg)) {
+            registers.add(reg);
+            FrameSlotKind kind;
+            switch (type) {
+                case I32:
+                    kind = FrameSlotKind.Int;
+                    break;
+                default:
+                    kind = FrameSlotKind.Illegal;
+                    break;
+            }
+            this.frameDescriptor.addFrameSlot(reg, kind);
+        }
+    }
+
+    private void getArguments() {
+        for (int i = 1; i < registers.size(); i++) {
```

Comment:
Could you use a named constant instead of `1`? If you skip the stack pointer argument you can use `LLVMCallNode.ARG_START_INDEX`.